### PR TITLE
Update to GNOME 42

### LINKED ISSUE #27: Update runtime to Gnome 40/41
just the title

### PR DESCRIPTION
**Pull request by hfiguiere**
Close #27 
Close #28 
Close #29

### CODE REVIEW

**Comment by hfiguiere** [comment]
@aleb @thiblahute can I merge this ?

**Comment by aleb** [comment]
I don't have time to test. I'll get back after we update the dev env.

**Comment by thiblahute** [comment]
Our [developement environment](https://developer.pitivi.org/HACKING.html?gi-language=undefined) in Pitivi is based on Flatpak and we should always have the flatpak manifest from there in sync with this one, starting by updating the one in the pitivi repo and then merging updating this on.

**Comment by hfiguiere** [comment]
The thing is that now 3.38 is deprecated and you never took the patch.

**Comment by thiblahute** [comment]
@aleb this should be good to go now right?

**Comment by aleb** [comment]
Not sure if there are issues with 40. The dev env is on 41 now.

**Comment by aleb** [comment]
Then we need to change the pygobject path in the json manifest.

**Comment by hfiguiere** [comment]
you already need to change the pre-built wheels because the python version got upgraded.

**Comment by hfiguiere** [comment]
Reworked it to update to GNOME 41:

- updated python package for 3.9 compatibility
- updated gstreamer to latest for the branch
- patched appdata for missing screenshot

**Comment by hfiguiere** [comment]
evidence that:
1. python package should be build from source
2. that I only built on aarch64

**Comment by hfiguiere** [comment]
bot, build

**Comment by hfiguiere** [comment]
This works here on x86_64

**Comment by hfiguiere** [comment]
Now with GNOME 42.

**Comment by hfiguiere** [comment]
build fail because libcanberra.

**Comment by hfiguiere** [comment]
this update uses libcanberra from shared-modules

**Comment by hfiguiere** [comment]
bot, build

**Comment by hfiguiere** [comment]
bot, build

**Comment by hfiguiere** [comment]
that build failure is back...

**Comment by A6GibKm** [comment]
I think that gst-plugging-base is in the runtime. See the error I am getting in my PR, the metainfo also needs patching since the screenshots urls do not exist anymore.

**Comment by hfiguiere** [comment]
> I think that gst-plugging-base is in the runtime. See the error I am getting in my PR, the metainfo also needs patching since the screenshots urls do not exist anymore.

Updated gstreamer and patched the appdata.

**Comment by hfiguiere** [comment]
This build is broken because of gst-python: python path did change. Will update.

File: pitivi-appdata.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/data/org.pitivi.Pitivi.appdata.xml.in b/data/org.pitivi.Pitivi.appdata.xml.in
+index f449d2a..f1a801d 100644
+--- a/data/org.pitivi.Pitivi.appdata.xml.in
++++ b/data/org.pitivi.Pitivi.appdata.xml.in
+@@ -21,7 +21,7 @@
+   </description>
+   <screenshots>
+     <screenshot type="default">
+-      <image>http://www.pitivi.org/i/screenshots/software_center_main.jpg</image>
++      <image>http://www.pitivi.org/i/screenshots/2020.05.png</image>
```
Comment (aleb) [inline pitivi-appdata.patch:10]: https

**Comment by aleb** [comment]
Can we just copy `shared-modules/libcanberra/libcanberra.json` over instead of using the shared-modules submodule? I'd rather never have to deal with git submodules again.

**Review by aleb — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
> Can we just copy `shared-modules/libcanberra/libcanberra.json` over instead of using the shared-modules submodule? I'd rather never have to deal with git submodules again.

And then it end up back to the same situation where things break and don't get updated. `shared-modules` are maintained properly and save a lot of time.

**Comment by A6GibKm** [comment]
> Can we just copy `shared-modules/libcanberra/libcanberra.json` over instead of using the shared-modules submodule? I'd rather never have to deal with git submodules again.

The point of using it is precisely so that modules are well maintained and not broken, note that this is precisely what happened here where it needed both a patch and a newer url.

**Comment by aleb** [comment]
The submodule also has to be updated, it does not update by itself, no? 

It's great that the shared-modules is well maintained, I'm all for copying what we need from there.

Am I missing anything?

Anyway, I don't even have merge rights in this repo, so don't mind me

**Comment by hfiguiere** [comment]
> The submodule also has to be updated, it does not update by itself, no?

Nothing updates by itself, but this is the path of least resistance.

> Anyway, I don't even have merge rights in this repo, so don't mind me. You seem to know what you are doing. 👍

Ok I'll merge then.

File: pitivi-appdata.patch (hunk 1 above)
Comment (hfiguiere) [reply pitivi-appdata.patch]: done

**Comment by hfiguiere** [comment]
Also added fallback-x11 as per #29

**Pull request merged**